Bearded.Utilities.Tests/Noise/PerlinNoiseTests.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using Bearded.Utilities.Noise;
+using Bearded.Utilities.Tests.Generators;
+using FluentAssertions;
+using FsCheck.Xunit;
+
+namespace Bearded.Utilities.Tests.Noise
+{
+    public sealed class PerlinNoiseTests
+    {
+        [Property(Arbitrary = new[] { typeof(DoubleGenerators.UnitIntervalUpperBoundExclusive) })]
+        public void GeneratesMapThatThrowsIfCoordinatesTooSmall(int seed, double x, double y)
+        {
+            var map = PerlinNoise.Generate(5, 5, seed);
```

Comment:
As explained in the PR description: the tests om this PR are by no means meant to be exhaustive.

In addition, it is hard to test that these parameters really do something. The randomness involved makes it hard to make hard assertions about the output. There is always a chance that the random output of a 10x10 grid just happens to line up really well with the properties of a 5x5 grid.

We can always generate two maps with the exact same seed with different dimensions and assume they are different in at least one point when converted to a discrete array, but there is no guarantee this is going to be the case for all seeds, and I think even that level of flakiness is not something we want to avoid in our tests.